Dntc.Common/Syntax/MethodBlock.cs

```diff
@@ -115,6 +126,7 @@ private static IReadOnlyList<CStatementSet> MoveDeclarationsToBeginning(IReadOnl
             }
         }
 
-        return declarations.Concat(nonDeclarations).ToArray();
+        // Combine in the order: initialLineStatements -> declarations -> other non-declarations
+        return initialLineStatements.Concat(declarations).Concat(nonDeclarations).ToArray();
```

Comment:
Is it worth keeping all the line statements?  I assume only the last one would be useful.

Reply:
fixed